components/yaml/src/main/java/datadog/yaml/YamlParser.java

```diff
@@ -1,19 +1,97 @@
 package datadog.yaml;
 
-import java.io.FileInputStream;
+import datadog.cli.CLIHelper;
 import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
 import org.yaml.snakeyaml.Yaml;
 
 public class YamlParser {
   // Supports clazz == null for default yaml parsing
   public static <T> T parse(String filePath, Class<T> clazz) throws IOException {
     Yaml yaml = new Yaml();
-    try (FileInputStream fis = new FileInputStream(filePath)) {
-      if (clazz == null) {
-        return yaml.load(fis);
-      } else {
-        return yaml.loadAs(fis, clazz);
+    String content = new String(Files.readAllBytes(Paths.get(filePath)));
```

Comment:
```suggestion
    String content = new String(Files.readAllBytes(Paths.get(filePath)), StandardCharsets.UTF_8);
```
this avoids using `new String(byte[])` which will use the platform encoding, which may not be UTF_8